activejob/lib/active_job/queue_adapters/resque_adapter.rb

```diff
@@ -27,6 +27,7 @@ module QueueAdapters
     #   Rails.application.config.active_job.queue_adapter = :resque
     class ResqueAdapter
       def enqueue(job) #:nodoc:
+        JobWrapper.instance_variable_set(:@queue, job.queue_name)
```

Comment:
Thank you for comment, but in [resque](https://github.com/resque/resque/blob/master/lib/resque.rb#L79), they make frequent use of `#constantize` method that needs `JobWrapper` class. So I decided to keep `JobWrapper` class singleton.